docs/src/index.md

```diff
@@ -268,6 +268,222 @@ df2 = @eachrow df begin
 end
 ```
 
+## Row-wise transformations with `@byrow`
+
+DataFrames.jl provides the function-wrapper `ByRow`. `ByRow(f)(x, y)`
+is roughly equivalent to `f.(x, y)`, with a few exceptions discussed below. 
+DataFramesMeta.jl allows users to construct expressions using `ByRow` 
+function wrapper with the syntax `@byrow`. 
+
+`@byrow` is not a "real" macro and cannot be used outside of 
+DataFramesMeta macros. However it's behavior within DataFramesMeta.jl
+macros should be indistinguishable from externally defined macros. 
+Thought of as a macro `@byrow` accepts a single argument and 
+creates an anonymous function wrapped in `ByRow`.  For example,
+
+```julia
+@transform(df, y = @byrow :x == 1 ? "true" : "false)
+```
+
+becomes
+
+```julia
+transform(df, :x => ByRow(x -> x == 1 ? "true", "false") => :y)
+```
+
+Macros that accept `@byrow`:
+
+* `@transform` and `@transform!`, `@select`, `@select!`, and `@combine`. `
+  @byrow` appears in the right hand side of expressions, of the form `@select(
+  df, z = @byrow :x * :y)`. 
+* `@where` and `@orderby`, with syntax of the form `@where(df, @byrow :x > :y)`
+* `@with`, where the anonymous function created by `@with` is wrapped in
+  `ByRow`. `@with df @byrow :x * :y`, which is conceptually similar to 
+  `map((x, y) -> x * y, df.x, df.y)`.
+
+To avoid writing `@byrow` multiple times when performing multiple
+operations, it is allowed `@byrow` at the beginning of a block of 
+operations. All transformations in the block will operate by row.
```

Comment:
one transformation is one expression?

Reply:
Yes, that's the new rule after #245. Hopefully it's not too confusing to people.